features/wallet/impl/src/main/java/com/tangem/feature/wallet/presentation/wallet/state/WalletContentItemState.kt

```diff
@@ -0,0 +1,13 @@
+package com.tangem.feature.wallet.presentation.wallet.state
+
+import com.tangem.feature.wallet.presentation.common.state.TokenItemState
+
+/**
+ * @author Andrew Khokhlov on 20/06/2023
+ */
+internal sealed interface WalletContentItemState {
+
+    data class NetworkGroupTitle(val text: String) : WalletContentItemState
+
+    data class Token(val tokenItemState: TokenItemState) : WalletContentItemState
+}
```

Comment:
Можно сделать так и можно будет убрать `when` в установке `id` элемента в списке

```suggestion
internal sealed interface WalletContentItemState {
    val id: String

    data class NetworkGroupTitle(val text: String) : WalletContentItemState {
       override val id = text
    }

    data class Token(val tokenItemState: TokenItemState) : WalletContentItemState {
        override val id = tokenItemState.id
    }
}
```